0000-ilp-ping-protocol/0000-ilp-ping-protocol.md

```diff
@@ -0,0 +1,78 @@
+
+## Terminology
+
+* The **Initiator** is the Interledger node initiating the ping request.
+* The **Recipient** is the Interledger node responding to the Initiator's request.
+* **Source amount** is the amount debited from the sender of an ILP payment.
+* **Destination amount** is the amount credited to the Recipient of an ILP payment.
+* The **Known Preimage** is the [ASCII](https://tools.ietf.org/html/rfc20) string `pingpingpingpingpingpingpingping`, which in binary is `70696E6770696E6770696E6770696E6770696E6770696E6770696E6770696E67`. This value is used in Unidirectional mode only. 
+* The **Known Condition** is the SHA256 hash of the Known Preimage, which when encoded using Base64 is `jAC8DGFPZPfh4AtZpXuvXFe2oRmpDVSvSJg2oT+bx34=`. This value is used in Unidirectional mode only.
+
+## Unidirectional Mode
+
+1. Initiator sends a ILP Prepare packet using the following details:
+   - **Destination**: Recipient's Interledger address.
+   - **Amount**: Any amount chosen by the Initiator.
+   - **Expiry**: Any appropriate expiry.
+   - **Condition**: The Known Condition, as defined in [Terminology](#terminology).
+   
+2. Upon receiving the Prepare packet, the Recipient MUST fulfill the payment using the following information:
+   - **Fulfillment**: The Known Fulfillment, as defined above.
+
+Note that a Recipient _MAY_ reject the payment if appropriate, for example due to an insufficient amount, invalid expiry, or for other factors.
+
+## Bidirectional Mode
+
+1. Initiator generates a new random 32-byte value to use as a Fulfilment (`F`) and generates the corresponding condition (`C`).
+2. Initiator sends an ILP Prepare packet to the Recipient with the following details:
+   - **Destination**: The Recipient's Interledger address.
+   - **Amount**: Any amount chosen by the Initiator.
+   - **Expiry**: An appropriate expiry that is long enough to allow for two round trips.
+   - **Condition**: The condition `C`.
+   - **Data**: A concatenated series of bytes with the following information:
+      - the bytes of the ASCII string `ECHOECHOECHOECHO`. In binary this value is `0x4543484F4543484F4543484F4543484F`.
+      - the byte `0x00`
+      - The ILP address of the Initiator (i.e., the return address) as an OER-encoded, variable length IA5 string.
+4. Upon receiving the Prepare packet, the Recipient identifies that it is a bidirectional Ping request by confirming the following:
+  1. The packet is addressed directly to itself (i.e. it has no tx or child suffix).
+  1. The packet has a payload conforming to the specification above (e.g., the byte following the `ECHOECHOECHOECHO` prefix in the payload is `0x00`).
+5. The Recipient DOES NOT immediately send an ILP Fulfill response. Instead the Recipient sends a new ILP `Prepare` packet addressed to the address parsed from the original original packet's payload. This new packet has following details:original 
```

Comment:
```suggestion
5. The Recipient DOES NOT immediately send an ILP Fulfill response. Instead the Recipient sends a new ILP `Prepare` packet addressed to the address parsed from the original original packet's payload. This new packet has following details: 
```

Reply:
This was fixed by the changes from @dora-gt